spec/unit/room.spec.ts

```diff
@@ -2231,6 +2231,84 @@ describe("Room", function () {
             });
             expect(room.getAvatarFallbackMember()?.userId).toBe(userD);
         });
+
+        it("should return undefined if the room is a 1:1 with a functional member", async function() {
+            const room = new Room(roomId, null!, userA);
+            // room.getMembers = jest.fn().mockReturnValue({
+            //     userB: new RoomMember(roomId, userB),
+            // });
+            jest.spyOn(Room.prototype as any, "getInvitedAndJoinedFunctionalMemberCount").mockReturnValue(1);
+            await room.currentState.setStateEvents([
+                utils.mkMembership({
+                    user: userA,
+                    mship: "join",
+                    room: roomId,
+                    event: true,
+                    name: "User A",
+                }),
+                utils.mkMembership({
+                    user: userB,
+                    mship: "join",
+                    room: roomId,
+                    event: true,
+                    name: "User B",
+                }),
+                utils.mkEvent({
+                    type: UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name,
+                    skey: "",
+                    room: roomId,
+                    event: true,
+                    content: {
+                        service_members: [userB],
+                    },
+                }),
+            ]);
+            expect(room.getAvatarFallbackMember()).toBeUndefined();
+        });
+
+        it("should use summary heroes of nonfunctional member if 1:1 plus functional member", async function() {
+            const room = new Room(roomId, null!, userA);
+            // room.getMembers = jest.fn().mockReturnValue({
+            //     userB: new RoomMember(roomId, userB),
+            // });
+            jest.spyOn(Room.prototype as any, "getInvitedAndJoinedFunctionalMemberCount").mockReturnValue(1);
+            await room.currentState.setStateEvents([
+                utils.mkEvent({
+                    type: UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name,
+                    skey: "",
+                    room: roomId,
+                    event: true,
+                    content: {
+                        service_members: [userB],
+                    },
+                }),
+            ]);
+            room.currentState.markOutOfBandMembersStarted();
+            room.currentState.setOutOfBandMembers([
```

Comment:
why do we use `...OutOfBandMembers` for this test rather than `setStateEvents`? Maybe we need a separate test to check the out-of-band-members case?

Reply:
Because I copied it from the test above :upside_down_face: I'm not sure what the difference is in the first place. What are they, and *do* I need to do something about them?

Reply:
I wish I knew... let's ignore them for now